<launch>
  <arg name="GRIPPER_CAM_SERIAL" value="128422272240" />
  <arg name="CUTTER_CAM_SERIAL" value="130322272760" />

  <!-- Gripper Camera -->
  <include file="$(find realsense2_camera)/launch/rs_camera.launch">
    <arg name="camera" value="gripper_cam" />
    <arg name="serial_no" value="$(arg GRIPPER_CAM_SERIAL)" />
    <arg name="enable_pointcloud" value="false" />
    <arg name="align_depth" value="true" />
    <arg name="color_width" value="848" />
    <arg name="color_height" value="480" />
    <arg name="color_fps" value="30" />
    <arg name="depth_width" value="848" />
    <arg name="depth_height" value="480" />
    <arg name="depth_fps" value="30" />
  </include>

<!-- Cutter Camera -->

  <include file="$(find realsense2_camera)/launch/rs_camera.launch">
    <arg name="camera" value="cutter_cam" />
    <arg name="serial_no" value="$(arg CUTTER_CAM_SERIAL)" />
    <arg name="enable_pointcloud" value="false" />
    <arg name="align_depth" value="true" />
    <arg name="color_width" value="848" />
    <arg name="color_height" value="480" />
    <arg name="color_fps" value="30" />
    <arg name="depth_width" value="848" />
    <arg name="depth_height" value="480" />
    <arg name="depth_fps" value="30" />
  </include>

  <arg name="pkg_path" value="$(find vader_perception)" />
  
  <!-- Then set the weights_path parameter with the resolved path -->
  <param name="fruit_weights_path" value="$(arg pkg_path)/weights/fruit_fvd.onnx" />
  <param name="fruit_drive_url" value="https://drive.google.com/uc?id=1nc8P4PtmbLeK7Y1u9ZnUYydurS6sc8h_" />
  <param name="peduncle_weights_path" value="$(arg pkg_path)/weights/peduncle_fvd.onnx" />
  <param name="peduncle_drive_url" value="https://drive.google.com/uc?id=1X4v8yjkXdBZJRr59IWfZ6JKWETy3N5-f" />


    <!-- Segmentation Confidence -->
  <param name="fruit_confidence" value="0.7" />
  <param name="peduncle_confidence" value="0.2" />                                                                                                                                                                                                                                                                                      
  
  
  <!-- Start the fruit detection node -->
  <node name="pose_estimation" pkg="vader_perception" type="dual_pose_estimation_node.py" output="screen">
  </node>

   <!-- Launch RViz with custom configuration -->
  <!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg pkg_path)/rviz_config/dual_cam.rviz" /> -->

  <!-- Launch UI -->
  <!-- <node name="robot_ui_node" pkg="vader_perception" type="user_interface.py" output="screen">
  </node> -->

</launch>

